### PR TITLE
Feature: Support for limiting the page size of fetchItemList

### PR DESCRIPTION
**Pull request by phil1995**
Introduce an optional maximum page size for `fetchItemList(forFolderAt:pageToken:)`. This is necessary to ensure that folders that contain a lot of items can also be opened in the `FileProviderExtension`, since it is only allowed to use a small amount of memory (<20MB). Therefore this PR is also related to the fix of issue [#232](https://github.com/cryptomator/ios/issues/232) in the `iOS` repo.

Since the LocalFileSystem as well as WebDAV do not provide for pagination, a `DirectoryContentCache` has been introduced. The cache gets filled with all items at the beginning of an initial, i.e. `pageToken = nil`, `fetchItemList(forFolderAt:pageToken:)` call and can mimic pagination via a page token.

Only pCloud does not provide any support for pagination. Therefore, there will still be problems with large folders in connection with the `PCloudCloudProvider`.
A possible alternative here would be a custom implementation of a chunk-based parser that also works with the `DirectoryContentCache`.

In addition, the `VaultFormat6Decorator` and `VaultFormat7Decorator` now use the `pageToken` when deleting a folder, since `fetchItemListExhaustively(forFolderAt:)` was previously used here, which keeps all items (in the form of `CloudItemMetadata` elements) of the folder in memory and can therefore also lead to memory limit problems.

### CODE REVIEW

File: Sources/CryptomatorCloudAccess/WebDAV/WebDAVProvider.swift [hunk 1]
```diff
@@ -7,14 +7,15 @@
 //
 
 import Foundation
+import GRDB
```
Comment (tobihagemann) [inline Sources/CryptomatorCloudAccess/WebDAV/WebDAVProvider.swift:10]:
```suggestion
```

Unused import, right?

File: Tests/CryptomatorCloudAccessTests/WebDAV/PropfindResponseCacheTests.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+//
+
+import GRDB
+import XCTest
+#if canImport(CryptomatorCloudAccessCore)
+@testable import CryptomatorCloudAccessCore
+#else
+@testable import CryptomatorCloudAccess
+#endif
+/*
```
Comment (tobihagemann) [inline Tests/CryptomatorCloudAccessTests/WebDAV/PropfindResponseCacheTests.swift:16]: What about these tests?

**Review by tobihagemann — CHANGES_REQUESTED** [review]

File: Sources/CryptomatorCloudAccess/WebDAV/WebDAVProvider.swift (hunk 1 above)
Comment (phil1995) [reply Sources/CryptomatorCloudAccess/WebDAV/WebDAVProvider.swift:10]: No, we need GRDB to initialize the `DatabaseQueue` (see line 43):
```swift
self.directoryContentCache = try DirectoryContentDBCache(dbWriter: DatabaseQueue(path: dbURL.path), maxPageSize: maxPageSize)
```

File: Tests/CryptomatorCloudAccessTests/WebDAV/PropfindResponseCacheTests.swift (hunk 2 above)
Comment (phil1995) [reply Tests/CryptomatorCloudAccessTests/WebDAV/PropfindResponseCacheTests.swift]: Ah thanks this were some early unit tests before we unified the two implementations as `DirectoryContentCache`.
Resolved by 5206564e96492d4c428fdc207e2f12e8f1fdb964.

**Review by tobihagemann — APPROVED** [review]

**Pull request merged**